fdm-app/app/routes/farm.$b_id_farm.$calendar.balance.organic-matter.$b_id.tsx

```diff
@@ -0,0 +1,336 @@
+    getOrganicMatterBalance,
+} from "@svenvw/fdm-calculator"
+import { getFarm, getField } from "@svenvw/fdm-core"
+import {
+    ArrowDownToLine,
+    ArrowRightLeft,
+    ArrowUpFromLine,
+    CircleAlert,
+    CircleCheck,
+} from "lucide-react"
+import { Suspense, use } from "react"
+import {
+    data,
+    type LoaderFunctionArgs,
+    type MetaFunction,
+    NavLink,
+    useLoaderData,
+    useLocation,
+} from "react-router"
+import { OrganicMatterBalanceChart } from "~/components/blocks/balance/organic-matter-chart"
+import OrganicMatterBalanceDetails from "~/components/blocks/balance/organic-matter-details"
+import { NitrogenBalanceFallback } from "~/components/blocks/balance/skeletons"
+import { Button } from "~/components/ui/button"
+import {
+    Card,
+    CardContent,
+    CardDescription,
+    CardFooter,
+    CardHeader,
+    CardTitle,
+} from "~/components/ui/card"
+import { getSession } from "~/lib/auth.server"
+import { getTimeframe } from "~/lib/calendar"
+import { clientConfig } from "~/lib/config"
+import { handleLoaderError, reportError } from "~/lib/error"
+import { fdm } from "~/lib/fdm.server"
+import { useCalendarStore } from "~/store/calendar"
+
+// Meta
+export const meta: MetaFunction = () => {
+    return [
+        {
+            title: `Organische Stof | Perceel | Nutriëntenbalans| ${clientConfig.name}`,
+        },
+        {
+            name: "description",
+            content: "Bekijk de organische stofbalans van dit perceel.",
+        },
+    ]
+}
+
+export async function loader({ request, params }: LoaderFunctionArgs) {
+    try {
+        const b_id_farm = params.b_id_farm
+        if (!b_id_farm) {
+            throw data("invalid: b_id_farm", { status: 400 })
+        }
+        const b_id = params.b_id
+        if (!b_id) {
+            throw data("invalid: b_id", { status: 400 })
+        }
+        const session = await getSession(request)
+        const timeframe = getTimeframe(params)
+        const farm = await getFarm(fdm, session.principal_id, b_id_farm)
+        if (!farm) {
+            throw data("not found: b_id_farm", { status: 404 })
+        }
+        const field = await getField(fdm, session.principal_id, b_id)
+
+        const organicMatterBalancePromise = collectInputForOrganicMatterBalance(
+            fdm,
+            session.principal_id,
+            b_id_farm,
+            timeframe,
+            b_id,
+        ).then(async (input) => {
+            const organicMatterBalanceResult = await getOrganicMatterBalance(
+                fdm,
+                input,
+            )
+            let fieldResult = organicMatterBalanceResult.fields.find(
+                (field: { b_id: string }) => field.b_id === b_id,
+            )
+
+            if (!fieldResult) {
+                throw new Error(
+                    `Organic matter balance data not found for field ${b_id}`,
+                )
+            }
+            if (fieldResult.errorMessage) {
```

Comment:
does this imply that if an error occurs, the page remains empty?

Reply:
No, we show than a card with an error message. For example if data of required soil parameters are missing we tell them to fill in those values. No tractor page is shown